src/engine.spec.ts

```diff
@@ -593,6 +593,58 @@ describe("Engine", () => {
       expect(engine.player(Player.Player1).data.advTechTiles).to.include(engine.advTechTiles[AdvTechTilePos.GaiaProject].tile);
       expect(engine.advTechTiles[AdvTechTilePos.GaiaProject].numTiles).to.equal(0);
     });
+
+    it("should allow Taklons to leech with +t freeIncome", () => {
+      const moves = parseMoves(`
+        init 2 randomSeed
+        p1 faction terrans
+        p2 faction taklons
+        p1 build m -4x-1
+        p2 build m -3x-2
+        p2 build m -6x3
+        p1 build m -4x2
+        p2 booster booster3
+        p1 booster booster4
+        p1 build ts -4x-1.
+        p2 leech 1pw
+        p2 build ts -3x-2.
+        p1 leech 2pw
+        p1 build ts -4x2.
+        p2 leech 1pw
+        p2 build PI -3x-2.
+        p1 leech 2pw
+        p1 build lab -4x-1. tech gaia.
+        p2 leech 3pw,1t
```

Comment:
The two test can be combined into one.

Remove that last line in the move list. Then:

```ts
expect(() => new Engine([...moves, "p2 leech 3pw,1t"])).to.not.throw();
expect(() => new Engine([...moves, "p2 leech 1t,3pw"])).to.not.throw();
```

Reply:
ok